knimin/lib/data_access.py

```diff
@@ -4258,6 +4284,84 @@ def delete_shotgun_plate(self, plate_id):
                 # plate_id did not exist
                 raise ValueError('Shotgun Plate %s does not exist' % plate_id)
 
+    def quantify_shotgun_plate(self, shotgun_plate_id, email, volume,
+                               plate_reader, plate_concentration):
+        """Adds the DNA quantification information to the shotgun plate
+
+        Parameters
+        ----------
+        shotgun_plate_id : int
+            The shotgun plate id
+        email : str
+            The email of the user
+        volume : float
+            The volume used for DNA quantification
+        plate_reader : str
+            The plate reader used
+        plate_concentration : 2d numpy array of floats
+            The per-well DNA concentration
+
+        Raises
+        ------
+        ValueError
+            If `plate_concentration` dimensions doesn't match the plate type
+        """
+        with TRN:
+            sgp = self.read_shotgun_plate(shotgun_plate_id)
+            rp = dict(self.read_plate_type(sgp['plate_type_id']))
+            pc_rows, pc_cols = plate_concentration.shape
+            processing_robot_id = self.get_or_create_property_option_id(
+                "plate_reader", plate_reader)
+
+            if pc_rows != rp['rows'] or pc_cols != rp['cols']:
```

Comment:
This "concantenation" is always 96 * 4 = 384 for now, but it's also compatible with any number x * 4 = 4x in the future, is that right?